app/src/main/java/org/stepic/droid/notifications/LocalReminderImpl.kt

```diff
@@ -152,6 +153,46 @@ class LocalReminderImpl
         remindAboutApp(null)
     }
 
+    private val registrationRemindHandling = AtomicBoolean(false)
+
+    override fun remindAboutRegistration() {
+        threadPoolExecutor.execute {
+            val isNotLoading = registrationRemindHandling.compareAndSet(false, true)
+            if (isNotLoading) try {
```

Comment:
it is tricky. I think it is good idea to avoid `if () try { `on 1 line. The better solution is extract body of try/finally in private method, and braces should not be omitted.